packages/interactivity/src/init.ts

```diff
@@ -44,14 +77,8 @@ export const init = async () => {
 		setTimeout( resolve, 0 );
 	} );
 
+	observedNodeCount = nodes.length;
 	for ( const node of nodes ) {
-		if ( ! hydratedIslands.has( node ) ) {
-			await splitTask();
-			const fragment = getRegionRootFragment( node );
-			const vdom = toVdom( node );
-			initialVdom.set( node, vdom );
-			await splitTask();
-			hydrate( vdom, fragment );
-		}
+		intersectionObserver.observe( node );
```

Comment:
Following up on your reply https://github.com/WordPress/gutenberg/pull/58284#discussion_r1906091159, why not do this?
```suggestion
		if ( hydratedIslands.has( node ) ) {
			observedNodeCount--;
			continue;
		}
		intersectionObserver.observe( node );
```